src/Visualization/Python/Render3D/Domain.py

```diff
@@ -0,0 +1,192 @@
+    zoom_factor,
+    camera_theta,
+    camera_phi,
+    clip_origin,
+    clip_normal,
+    slice,
+):
+    """Renders a 3D domain with elements and grid lines
+
+    This rendering is a starting point for visualizations of the domain
+    geometry, e.g. for publications.
+
+    XMF_FILE is the path to the XMF file that references the simulation data.
+    It is typically generated by the 'generate-xdmf' command.
+    You can also provide a second XMF file with higher resolution data, which
+    is used to render the outlines of elements to make them smoother.
+    """
+    import paraview.simple as pv
+
+    # Load data
+    volume_data = pv.XDMFReader(
+        registrationName="VolumeData", FileNames=[xmf_file]
+    )
+    if hi_res_xmf_file:
+        hi_res_volume_data = pv.XDMFReader(
+            registrationName="HiResVolumeData", FileNames=[hi_res_xmf_file]
+        )
+
+    render_view = pv.GetActiveViewOrCreate("RenderView")
+    render_view.UseLight = 0
+    render_view.UseColorPaletteForBackground = 0
+    render_view.Background = 3 * [1.0]
+    render_view.OrientationAxesVisibility = 0
+
+    def slice_or_clip(triangulate, **kwargs):
+        if slice:
+            result = pv.Slice(
+                **kwargs,
+                SliceType="Plane",
+                Triangulatetheslice=triangulate,
+            )
+            result.SliceType.Origin = clip_origin
+            result.SliceType.Normal = clip_normal
+        else:
+            result = pv.Clip(**kwargs, ClipType="Plane")
+            result.ClipType.Origin = clip_origin
+            result.ClipType.Normal = clip_normal
+        return result
+
+    # Show grid
+    grid = slice_or_clip(
+        registrationName="Grid", Input=volume_data, triangulate=False
+    )
+    grid_display = pv.Show(grid, render_view)
+    grid_display.Representation = "Surface With Edges"
+    grid_display.LineWidth = 1.0
+    grid_display.EdgeColor = 3 * [0.6]
+    # The following line works around a failure in `pv.ColorBy(..., None)`
+    grid_display.ColorArrayName = ("POINTS", None)
+    pv.ColorBy(grid_display, None)
+
+    # Show outline
+    outline_data = hi_res_volume_data if hi_res_xmf_file else volume_data
+    outline = slice_or_clip(
+        registrationName="Outline", Input=outline_data, triangulate=True
+    )
+    outline_display = pv.Show(outline, render_view)
+    outline_display.Representation = "Feature Edges"
+    outline_display.LineWidth = 3.0
+    outline_display.DiffuseColor = 3 * [0.0]
+    outline_display.AmbientColor = 3 * [0.0]
+    # The following line works around a failure in `pv.ColorBy(..., None)`
+    outline_display.ColorArrayName = ("POINTS", None)
+    pv.ColorBy(outline_display, None)
+
+    # Set resolution
+    layout = pv.GetLayout()
+    layout.SetSize(1200, 1200)
+
+    # Configure camera
+    camera = pv.GetActiveCamera()
+    camera.SetFocalPoint(*clip_origin)
+    camera.SetPosition(*(np.array(clip_origin) + np.array(clip_normal) * 10.0))
```

Comment:
This factor of 10 seems a bit arbitrary. I could imagine this not working for a BBH with an outer boundary of like 1000.